### PR TITLE
Event Parameters Networking

### PR DESCRIPTION
**Pull request by jessetilro**
EventParameters can now be submitted client-side and validated server-side.

Developed (and should therefore be tested) in parallel with https://github.com/jessetilro/pixelperfect-client/pull/25.

Resolves Issue #220.

### CODE REVIEW

**Comment by FlorisXll** [comment]
The code looks really good, haven't mannually tested it myself.

File: src/test/java/nl/tudelft/pixelperfect/client/ServerListenerTest.java [hunk 1]
```diff
@@ -74,10 +74,10 @@ public void testMessageReceived() {
     // Stubbing
     when(mockedMessage.getLabel()).thenReturn("Fire Event");
 
-    // Execution
```
Comment (dalderliesten) [inline src/test/java/nl/tudelft/pixelperfect/client/ServerListenerTest.java]: As a side note, do we have a "no spaces after comments" in our checkstyle?  It should not be happening, and I'm seeing it in a lot of places.

**Comment by dalderliesten** [comment]
Looks fine to me.

File: src/main/java/nl/tudelft/pixelperfect/event/EventLog.java [hunk 2]
```diff
@@ -93,26 +113,32 @@ public synchronized void replace(ArrayList<Event> log) {
...
-   *           The id used to find the event in the log.
+   * 
+   * @param id
+   *          The id used to find the event in the log.
    */
-  public synchronized void complete(int identity) {
-    for (int t = 0; t < events.size(); t++) {
-      if (events.get(t).getId() == identity) {
-        events.remove(t);
+  public synchronized void complete(int id, Collection<EventParameter> parameters) {
```
Comment (wouterz) [inline src/main/java/nl/tudelft/pixelperfect/event/EventLog.java]: @param parameters is still missing

**Comment by wouterz** [comment]
rest looks good to me

**Comment by jessetilro** [comment]
@wouterz You are right, I was rushing to get this functionality implemented since the deadline approaches and we need this to proceed the implementation of minigames. Therefore I didn't even check te reports. I will do so this afternoon.

**Comment by jessetilro** [comment]
Clean and ready to merge @wouterz.

**Comment by wouterz** [comment]
Good work!

**Pull request merged**